### DISCUSSION TITLE
does microstream need special type handlers for jdk 21?

### DISCUSSION THREAD

**Discussion by guykatz** [microstream-one/microstream#658] [Q&A] [answered]
hi all;
does microstream need a specilized type handlers for jdk 21?
if yes, when is this planned? and what are my limitations working with J21 and current MS
thanks!

**Comment by hg-ms** [comment, marked as answer]
Hi,
Regarding specialized type handlers you may still add the microstream-persistence-binary-jdk17 module to add support for the immutable list and set.
There are no specific type handlers for java 21 now.
Currently we don’t know about any java 21 specific issues.
